src/package.coffee

```diff
@@ -506,15 +506,18 @@ class Package
     @configSchemaRegisteredOnActivate = false
     @deactivateResources()
     @deactivateKeymaps()
+    result = Promise.resolve()
     if @mainActivated
       try
-        @mainModule?.deactivate?()
-        @mainModule?.deactivateConfig?()
-        @mainActivated = false
-        @mainInitialized = false
+        result = Promise.resolve(@mainModule?.deactivate?()).then =>
+          @mainModule?.deactivateConfig?()
+          @mainActivated = false
+          @mainInitialized = false
       catch e
```

Comment:
This `catch` block won't catch errors thrown from `deactivate()`. You'd need a `.catch` on the Promise to handle async errors instead:

```coffee
result = Promise.resolve(@mainModule?.deactivate?()).then =>
  @mainModule?.deactivateConfig?()
  @mainActivated = false
  @mainInitialized = false
result = result.catch e =>
  # Async errors from the promise
  console.error "Error deactivating package '#{@name}', e.stack
```